### ISSUE TITLE
Allow circuits to follow the cursor when pasting

### ISSUE THREAD

**Issue by YizhePKU** [CircuitVerse/CircuitVerse#971]
## Description

The current pasting behavior is to directly put down the circuits wherever the cursor is. This can lead to inconvenience and sometimes frustration.

* Often times the circuits don't land where the user wants them, so the user needs to move them manually.
* To make matters worse, the pasted circuits are not auto-selected, so the user needs to re-select whatever have been put down. This can be a nightmare when doing large-scale copy-pasting operations(as seen in #935, point 9 and 10).

## Proposed solution

Instead of putting down the circuits directly when the user hits Ctrl+V, let the circuits hover and follow the cursor until the user left clicks to place them down. This approach has several advantages:

* **Intuitive interface**. No need to figure out where to place the cursor before Ctrl+V.
* **Simple implementation**. Adding new circuits already behaves this way(newly added circuit will follow the cursor until left click). We only need to generalize that behavior to any group of circuits.

## Alternatives

Here is a less "radical" solution: After Ctrl+V we deselect whatever was selected before, and auto-selected the circuits that were put down. This makes it easier to move the circuits manually. But it's not as convenient.

I plan to work on this issue, and I would like to hear some feedback :)

**Comment by KaustubhDamania** [comment]
Is this issue still open? I'd like to work on it.

**Comment by YizhePKU** [comment]
@KaustubhDamania Yes. I planned to work on this myself but got distracted recently. You are welcome to try.

**Comment by KaustubhDamania** [comment]
Can anyone tell me where is the code for drawing new circuits or circuit elements in the simulator canvas? For e.g., when the circut element is copy pasted, a new element is drawn on the canvas. However I'm not able to find the code for the same in the paste function in public/js/logix.js.

**Comment by satu0king** [comment]
@YizhePKU, All the CAD tools I have used do not seem to do that. Copy means copy unless pasted. Because we might want to paste in this circuit, another subcircuit or entirely another project. The original issue posted by @DevL0rd actually says not to deselect everything once pasted. This means the pasted items needs to be selected, so it can be moved to the exact location after pasting. I see this is what you have written in the alternative. And I strongly propose we go with this as this is standard in CAD.

**Comment by KaustubhDamania** [comment]
@satu0king what if we display a toast message that an element(s) has been copied? So the user wouldn't be confused whether the element is copied or not.

**Comment by satu0king** [comment]
@KaustubhDamania you can show but tell me which CAD tool shows that elements have been copied? In this case it won't confuse the user further but as a thumb rule deviating from what is conventionally done is not a good idea.

**Comment by KaustubhDamania** [comment]
@satu0king yeah, got it! :+1:

**Comment by YizhePKU** [comment]
@satu0king The tool I'm using is LTSpice and it handles copy-pasting just like I've described. But yeah perhaps it is not typical, and we should aim for something that is more familar to the users.

**Comment by YizhePKU** [comment]
@satu0king I agree that we should aim for behavior that is familiar to most users. But there might be a confusion here. What I meant is that copying is still the same as before, but when pasting, let the circuits follow the cursor. This way we can still paste in another subcircuit or another project easily. That's how LTSpice works and I find it intuitive. Perhaps it is a solution worth considering.

I'm OK if we end up using the alternative solution though, as it seems to be the standard of CAD tools.

**Comment by Me-Priyank** [comment]
Outdated. Simulator has been moved to cv-frontend-vue repo
